x/evm/watcher/ac_querier.go

```diff
@@ -0,0 +1,263 @@
+	if v, ok := aq.p.Get(key); ok {
+		if v == nil || v.GetType() == TypeDelete { // for del key
+			return 0, nil
+		}
+		var height []byte
+		if rsp, ok := v.(*MsgLatestHeight); ok {
+			height = []byte(rsp.GetValue())
+		} else if value, ok := v.(*Batch); ok { // maybe v is from the dds
+			height = value.Value
+		}
+		h, e := strconv.Atoi(string(height))
+		return uint64(h), e
+	}
+	return 0, errACNotFound
+}
+
+func (aq *ACProcessorQuerier) GetBlockHashByNumber(key []byte) (common.Hash, error) {
+	if v, ok := aq.p.Get(key); ok {
+		if v == nil || v.GetType() == TypeDelete { // for del key
+			return common.Hash{}, nil
+		}
+		if rsp, ok := v.(*MsgBlockInfo); ok {
+			return common.HexToHash(rsp.GetValue()), nil
+		} else if value, ok := v.(*Batch); ok { // maybe v is from the dds
+			return common.HexToHash(string(value.Value)), nil
+		}
+	}
+	return common.Hash{}, errACNotFound
+}
+
+func (aq *ACProcessorQuerier) GetBlockHash(key []byte) (common.Hash, error) {
+	if v, ok := aq.p.Get(key); ok {
+		if v == nil || v.GetType() == TypeDelete { // for del key
+			return common.Hash{}, nil
+		}
+		if rsp, ok := v.(*MsgBlockInfo); ok {
+			return common.HexToHash(rsp.GetValue()), nil
+		} else if value, ok := v.(*Batch); ok { // maybe v is from the dds
+			return common.HexToHash(string(value.Value)), nil
+		}
+	}
+	return common.Hash{}, errACNotFound
+}
+
+func (aq *ACProcessorQuerier) GetCode(key []byte) ([]byte, error) {
+	if v, ok := aq.p.Get(key); ok {
+		if v == nil || v.GetType() == TypeDelete { // for del key
+			return nil, nil
+		}
+		if rsp, ok := v.(*MsgCode); ok {
+			return []byte(rsp.GetValue()), nil
+		} else if value, ok := v.(*Batch); ok { // maybe v is from the dds
+			return value.Value, nil
+		}
+	}
+	return nil, errACNotFound
+}
+
+func (aq *ACProcessorQuerier) GetCodeByHash(key []byte) ([]byte, error) {
+	if v, ok := aq.p.Get(key); ok {
+		if v == nil || v.GetType() == TypeDelete { // for del key
+			return nil, nil
+		}
+		if rsp, ok := v.(*MsgCodeByHash); ok {
+			return []byte(rsp.GetValue()), nil
+		} else if value, ok := v.(*Batch); ok { // maybe v is from the dds
+			return value.Value, nil
+		}
+	}
+	return nil, errACNotFound
+}
+
+func (aq *ACProcessorQuerier) GetStdTxHash(key []byte) ([]byte, error) {
+	if v, ok := aq.p.Get(key); ok {
+		if v == nil || v.GetType() == TypeDelete { // for del key
+			return nil, nil
+		}
+		if rsp, ok := v.(*MsgBlockStdTxHash); ok {
+			return []byte(rsp.GetValue()), nil
+		} else if value, ok := v.(*Batch); ok { // maybe v is from the dds
+			return value.Value, nil
+		}
+	}
+	return nil, errACNotFound
+}
+
+func (aq *ACProcessorQuerier) GetAccount(key []byte) (*types.EthAccount, error) {
+	if v, ok := aq.p.Get(key); ok {
+		if v == nil || v.GetType() == TypeDelete { // for del key
+			return nil, nil
+		}
+		if rsp, ok := v.(*MsgAccount); ok {
+			cp := rsp.account.Copy() // the account value will be modify by caller
```

Comment:
the comment `be modify` to `be modified`

Reply:
> the comment `be modify` to `be modified`

ok